src/libaktualizr/crypto/p11engine.cc

```diff
@@ -145,7 +145,25 @@ boost::filesystem::path P11Engine::findPkcsLibrary() {
 }
 
 PKCS11_SLOT* P11Engine::findTokenSlot() const {
-  PKCS11_SLOT* slot = PKCS11_find_token(ctx_.get(), wslots_.get_slots(), wslots_.get_nslots());
+  unsigned int i, nslot;
+  PKCS11_SLOT* slot = NULL;
+  PKCS11_SLOT* iterslot = NULL;
+  PKCS11_TOKEN* tok;
+  nslot = wslots_.get_nslots();
+
+  if (label_.empty()) {
+    // if label is not defined, aktualizr won't use any slots
+    LOG_ERROR << "Token label missing. Unable to select slot.";
+    return nullptr;
+  }
+  for (i = 0, iterslot = wslots_.get_slots(); i < nslot; i++, iterslot++) {
```

Comment:
Just minor. We may wanna define  `iterslot` before the loop `auto iterslot{wslots_.get_slots()}` and then define `i` within the loop `for (unsigned int i; i < nslot...`.